File: AOJ/0432/0432.cpp
#include<iostream>
using namespace std;
int N;
string G[1000], T[1000];

void rotate(){
  for ( int i = 0; i < N; i++ )
    for ( int j = 0; j < N; j++ )
      T[j][N-i-1] = G[i][j];
  for ( int i = 0; i < N; i++ ) G[i] = T[i];
}

int main(){
  int Q;
  int cnt = 4 * 100000;
  cin >> N;
  for ( int i = 0; i < N; i++ ) {
    cin >> G[i];
    T[i] = G[i];
  }
  cin >> Q;
  for ( int i = 0; i < Q; i++ ){
    int q; cin >> q; cnt += q;
  }
  cnt %= 4;
  for ( int i = 0; i < cnt; i++ ) rotate();
  for ( int i = 0; i < N; i++ ) cout << G[i] << endl;
  
  return 0;
}

File: AOJ/0432/0432.py
import copy

def rotate():
    T = copy.deepcopy(G)
    for i in range(N):
        for j in range(N):
            G[j][N-i-1] = T[i][j]

N = int(input())
G = [[' ' for _ in range(N)] for _ in range(N)]

for i in range(N):
    row = input()
    for j in range(N):
        G[i][j] = row[j]

cnt = 4 * 100000
Q = int(input())
for i in range(Q):
    q = int(input());
    cnt += q
cnt %= 4
for i in range(cnt): rotate()

for i in range(N): 
    for j in range(N): print(G[i][j], end='')
    print()
